strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/LoggingManagementController.java

```diff
@@ -238,11 +237,111 @@ public ResponseEntity uploadLogbackConfiguration(HttpServletRequest request,
         catch (IOException | LoggingConfigurationException e)
         {
             String message = "Could not upload logback configuration.";
+            
             logger.error(message, e);
 
-            return ResponseEntity.status(BAD_REQUEST)
-                                 .body(getResponseEntityBody(message, accept));
+            return ResponseEntity.status(BAD_REQUEST).body(getResponseEntityBody(message, accept));
         }
     }
 
+    @ApiOperation(value = "Used to get log directory.")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "The log directory was retrieved successfully."),
+                            @ApiResponse(code = 400, message = "Could not download log directory.") })
+    @GetMapping(value = { "/logs",
+                          "/logs/**" },
+                produces = TEXT_PLAIN_VALUE)
+    public void generateLogDirectoryListing(HttpServletRequest request,
+                                            HttpServletResponse response)
+            throws IOException
+    {
+        String requestUriString = request.getRequestURI();
+        String uriLogDirPath = requestUriString.replace("/logging", "");
+        Path localLogDirPath = Paths.get(PropertyUtils.getVaultDirectory(), uriLogDirPath);
+        
+        if (Files.notExists(localLogDirPath))
+        {
+            response.sendError(404, "File not found. Variables: " + uriLogDirPath + " " + localLogDirPath.toString());
+        }
+        
+        File file = localLogDirPath.toFile();
+        
+        //Sends a redirect if the URI does not end with a "/"
+        if (!requestUriString.endsWith("/"))
+        {
+            try
+            {
+                response.sendRedirect(requestUriString + "/");
+            }
+            catch (IOException e)
+            {
+                logger.debug("Error redirecting to " + requestUriString + "/");
+            }
+        }
+        
+        try
+        {
+            logger.debug(" browsing: " + localLogDirPath.toString());
+            
+            //Generating the HTML View
+            StringBuilder sb = new StringBuilder();
+            sb.append("<html>");
+            sb.append("<head>");
+            sb.append("<style>body{font-family: \"Trebuchet MS\", verdana, lucida, arial, helvetica, sans-serif;} table tr {text-align: left;}</style>");
+            sb.append("<title>Index of " + uriLogDirPath + "</title>");
+            sb.append("</head>");
+            sb.append("<body>");
+            sb.append("<h1>Index of " + uriLogDirPath + "</h1>");
+            sb.append("<table cellspacing=\"10\">");
+            sb.append("<tr>");
+            sb.append("<th>Name</th>");
+            sb.append("<th>Last modified</th>");
+            sb.append("<th>Size</th>");
+            sb.append("<th>Description</th>");
+            sb.append("</tr>");
+            sb.append("<tr>");
+            sb.append("<td colspan=4><a href=\"..\">..</a></td>");
```

Comment:
It might produce same bug as #572 . Maybe should add something like #603 ?

Reply:
Good note. I'll update the commit!